### DISCUSSION TITLE
Dynamic attributes with context objects

### DISCUSSION THREAD

**Discussion by Anthony-DE-FARIA** [wrabit/django-cotton#135] [Ideas]
Hey @wrabit,

What do you think of this idea? Do you find it relevant or even possible?

Before:
```
<!-- context = { 'today': Weather.objects.get(...) } -->
<c-weather :today="today"></c-weather>
```

After:
```
<!-- context = { 'today': Weather.objects.get(...) } -->
<c-weather :today="{{ today }}"></c-weather>
```

This change helps to quickly identify the context object in Cotton components and also enables auto-completion.

Cheers! 👍🏻

**Comment by wrabit** [comment]
@Anthony-DE-FARIA I get your point. I didn't consider it before but to me, it breaks some protocol in my mind because as soon as you `{{ ... }}` it's rendering something there and then, which would normally print out the dict/list/object to the html (as a string) and that would not be much use to the underlying component that wants to use it's attributes as a dict/list/object. We're actually wanting to send the variable by it's reference so the component can manage it how it wants.

What I did consider is if we allow dropping the quotes and `:` to be more django-esque i.e. like with `{% with %}`

```html
<c-weather today=today></c-weather>
```

I'm not entirely sold on that, what are your thoughts about that?

**Comment by mattbha** [comment]
personally I'd agree with @wrabit and avoid the {{ }} syntax as my mind goes to "rendering a variable". i guess it might technically 'still work' as the rendered variable could be then passed. But I think of the case where it'd a django ORM object would happen quite frequently and not work great in that regard.


I'd vote in favor of `:` being syntax for django-cotton to mean "pass this python variable directly" almost like a parameter in a function

**Comment by Anthony-DE-FARIA** [comment]
Yep, after using dj-cotton again and again for the past weeks I agree with you guys.

Named slots are doing the job perfectly if I need to introduce {{ today }} for context clarity for example.
